src/navigation/BottomTabNavigator/CustomTabBarContainer.js

```diff
@@ -21,6 +22,23 @@ const CustomTabBarContainer = ( { navigation }: Props ): Node => {
   const currentUser = useCurrentUser( );
   const [activeTab, setActiveTab] = useState( OBS_LIST_SCREEN_ID );
   const isDrawerOpen = useDrawerStatus() === "open";
+  const route = getCurrentRoute();
+  const currentRoute = route?.name || "";
+  useEffect( () => {
```

Comment:
Sorry for the late review.
This is not a good use of an useEffect call. We have not been too strict with that ourselves but will try to be more strict about that in future. useEffects are primarily for handling changes that need to happen because something outside React has changed, like on the server for example.

Please, check out his documentation here and refactor this code accordingly:
https://react.dev/learn/you-might-not-need-an-effect
https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes

Reply:
@jtklein i have made the changes